lib/Parse/Parser.cpp

```diff
@@ -1096,8 +1107,12 @@ Decl *Parser::ParseFunctionDefinition(ParsingDeclarator &D,
       TemplateInfo.Kind == ParsedTemplateInfo::Template &&
       Actions.canDelayFunctionBody(D)) {
     MultiTemplateParamsArg TemplateParameterLists(*TemplateInfo.TemplateParams);
-    
-    ParseScope BodyScope(this, Scope::FnScope|Scope::DeclScope);
+    // Checked C - consider checked function definition
+    ParseScope BodyScope(this,
+                         Scope::FnScope | Scope::DeclScope |
+                             (D.getDeclSpec().isCheckedSpecified() || isChecked
+                                  ? Scope::CheckedScope
```

Comment:
It looks this computation is repeated several times below (I am reviewing this via the GitHub interface which hides lines of code, so I might be wrong).   Instead of repeating the computation, please compute it once and store the result in a local variable.